Classes/System/Configuration/ConfigurationManager.php

```diff
@@ -86,8 +88,13 @@ public function getTypoScriptConfiguration(?int $contextPageId = null, int $cont
             $language = $site->getLanguageById($contextLanguageId);
             // @todo: Storage-Folder can not be used to get TypoScript Config!!!
             $uri = $site->getRouter()->generateUri($contextPageId, ['_language' => $language]);
+            $pageInformation = new PageInformation();
+            $pageInformation->setId($contextPageId);
+            $pageInformation->setPageRecord(BackendUtility::getRecord('pages', $contextPageId));
+            $pageInformation->setContentFromPid($contextPageId);
```

Comment:
Site is available here to.
Maybe extracting to get method and use it for 114-117 as well?

Reply:
no, here we have the concrete Page ID (`$contextPageId`), so we should not use the rootPage of the Site.